src/app/rpmostree-ex-builtin-jigdo2commit.c

```diff
@@ -188,67 +171,17 @@ impl_jigdo2commit (RpmOstreeJigdo2CommitContext *self,
   if (!treespec)
     return FALSE;
 
+  /* We're also "pure" jigdo - this adds assertions that we don't depsolve for example */
+  rpmostree_context_set_is_pure_jigdo (self->ctx);
   if (!rpmostree_context_setup (self->ctx, NULL, NULL, treespec, cancellable, error))
     return FALSE;
-  if (!rpmostree_context_download_metadata (self->ctx, cancellable, error))
+  if (!rpmostree_context_prepare (self->ctx, cancellable, error))
```

Comment:
Hmm, let's just make this a separate `prepare_jigdo` function? And then just `g_assert` at the top of both functions to make sure they're not mixed?

Reply:
Good suggestion, though I did the first but not the second; I *think* that down the line we might be able to get away with using a single `RpmOstreeContext*` for both the jigdo and layering changes?  In that case then we'd just call `rpmostree_context_prepare()`.